src/QuadForm/ClosestVector.jl

```diff
@@ -1,65 +1,55 @@
 import AbstractAlgebra.Generic: FreeModuleElem
 
 @doc Markdown.doc"""
-    convert_type(Q::MatrixElem, K::MatrixElem, d::RingElement) -> ZLat, Vector{RingElement}, RingElement
+    convert_type(Q::fmpq_mat, K::Vector{fmpq}, d::fmpq) -> ZLat, Vector{fmpq}, fmpq
 
-  Converts a quadratic triple QT = [Q, K, d] to the input values required for closest vector problem (CVP). 
+Converts a quadratic triple QT = [Q, K, d] to the input values required for closest vector problem (CVP). 
 """
-function convert_type(G::MatrixElem, K::MatrixElem, d::RingElement)
-    @assert typeof(G) == typeof(K) "Input matrices should be of the same type"
+function convert_type(G::fmpq_mat, K::Vector{fmpq}, d::fmpq)
```

Comment:
This achieves type stability by sacrificing generality. If that's what you want... but it's not strictly necessary , there are other ways.

actually you also change here and elsewhere some arguments from (single row/column) matrices to Vector...

Reply:
Hello, what would be another way of doing this? I am not very familiar with other methods. Could you please guide me?